pkg/sql/opt/cat/table.go

```diff
@@ -143,6 +143,11 @@ type Table interface {
 	// IsPartitionAllBy returns true if this is a PARTITION ALL BY table. This
 	// includes REGIONAL BY ROW tables.
 	IsPartitionAllBy() bool
+
+	// IsRefreshRequired returns true if the table is a materialized view
+	// created with the NO DATA option and is yet to be refreshed. Accessing
+	// such a view prior to running refresh returns an error.
+	IsRefreshRequired() bool
```

Comment:
Perhaps give this method the same name as `IsRefreshViewRequired`, or vice-versa? It means the same thing, after all.